chain/network/src/peer_store.rs

```diff
@@ -28,23 +54,32 @@ impl PeerStore {
         boot_nodes: &[PeerInfo],
     ) -> Result<Self, Box<dyn std::error::Error>> {
         let mut peer_states = HashMap::default();
+        let mut addr_peers = HashMap::default();
         for (key, value) in store.iter(ColPeers) {
             let key: Vec<u8> = key.into();
             let value: Vec<u8> = value.into();
             let peer_id: PeerId = key.try_into()?;
             let mut peer_state: KnownPeerState = value.try_into()?;
+            // Mark loaded node last seen to now, to avoid deleting them as soon as they are loaded.
+            peer_state.last_seen = to_timestamp(Utc::now());
             match peer_state.status {
                 KnownPeerStatus::Banned(_, _) => {}
                 _ => peer_state.status = KnownPeerStatus::NotConnected,
             };
+            if let Some(peer_addr) = peer_state.peer_info.addr.clone() {
+                addr_peers.insert(peer_addr, VerifiedPeer::new(peer_state.peer_info.id.clone()));
+            }
             peer_states.insert(peer_id, peer_state);
         }
         for peer_info in boot_nodes.iter() {
             if !peer_states.contains_key(&peer_info.id) {
+                if let Some(peer_addr) = peer_info.addr.clone() {
+                    addr_peers.insert(peer_addr, VerifiedPeer::new(peer_info.id.clone()));
```

Comment:
Yes, you are right, they should be signed, so they are not tampered. Otherwise if we are not connected to boot_nodes, other nodes we try to connect to can claim having peer_id from the boot_nodes and we change it.

Reply:
To fix it I added first boot nodes pairs as Signed connections, and then added other nodes stored on PeerStore as indirect connections. If there is a collision between a boot node and a new node, boot node pair will have preference.

Reply:
Makes sense